airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/FlushWorkers.java

```diff
@@ -41,18 +38,20 @@
 @Slf4j
 public class FlushWorkers implements AutoCloseable {
 
+  public static final long TOTAL_QUEUES_MAX_SIZE_LIMIT_BYTES = (long) (Runtime.getRuntime().maxMemory() * 0.8);
```

Comment:
Non-blocking, how come we're moving this from BufferManager here?

Reply:
yeah. i think this is a symptom of something leaking in the abstraction. i think trying to do a `flushall` to handle memory pressure is actually the root cause of the issue. if we handle that differently, then this wouldn't need to be here.

Reply:
that is right! maybe we can move this to proper constants class. can do later.